docs/monitoring.md

```diff
@@ -684,9 +688,256 @@ code in your Spark package**_. For sbt users, set the
 the `-Pspark-ganglia-lgpl` profile. In addition to modifying the cluster's Spark build
 user applications will need to link to the `spark-ganglia-lgpl` artifact.
 
-The syntax of the metrics configuration file is defined in an example configuration file,
+The syntax of the metrics configuration file and the parameters available for each sink are defined
+in an example configuration file,
 `$SPARK_HOME/conf/metrics.properties.template`.
 
+When using configuration parameters instead of the metrics configuration file, the relevant
```

Comment:
Maybe we could put full of explanation either here or `metrics.properties.template`. Since most of explanation is already going through `metrics.properties.template` so that might be better place to add explanation, but it's just my 2 cents.

After that, for here I guess we just need to explain end users need to put `spark.metrics.conf.` in addition to actual configuration name.